commit 3edda3c5e71d634efd8e1e849328a68fb9da37de
Author: Nicholas Marriott <nicholas.marriott@gmail.com>
Date:   Fri Jun 3 08:09:16 2022 +0000

    Do not unintentionally turn off all mouse mode when button is also present.

diff --git a/tty.c b/tty.c
index ea10e61eb6..49cf9795f7 100644
--- a/tty.c
+++ b/tty.c
@@ -814,7 +814,7 @@ tty_update_mode(struct tty *tty, int mode, struct screen *s)
 			tty_puts(tty, "\033[?1006h");
 		if (mode & MODE_MOUSE_ALL)
 			tty_puts(tty, "\033[?1000h\033[?1002h\033[?1003h");
-		if (mode & MODE_MOUSE_BUTTON)
+		else if (mode & MODE_MOUSE_BUTTON)
 			tty_puts(tty, "\033[?1000h\033[?1002h");
 		else if (mode & MODE_MOUSE_STANDARD)
 			tty_puts(tty, "\033[?1000h");
